lighthouse-core/lib/web-inspector.js

```diff
@@ -7,64 +7,55 @@
 'use strict';
 
 /**
- * Stubbery to allow portions of the DevTools frontend to be used in lighthouse. `WebInspector`
+ * Stubbery to allow portions of the DevTools frontend to be used in lighthouse. `SDK`
  * technically lives on the global object but should be accessed through a normal `require` call.
  */
 module.exports = (function() {
-  if (global.WebInspector) {
-    return global.WebInspector;
-  }
-
-  // Global pollution.
-  // Check below is to make it worker-friendly where global is worker's self.
-  if (global.self !== global) {
-    global.self = global;
+  if (global.SDK) {
+    return global.SDK;
   }
 
+  // Dependencies for effective CSS rule calculation. Global pollution!
+  global.SDK = {};
+  global.TextUtils = {};
   global.Node = {
     ELEMENT_NODE: 1,
     TEXT_NODE: 3,
   };
-
-  global.CSSAgent = {};
-  global.CSSAgent.StyleSheetOrigin = {
-    Injected: 'injected',
-    UserAgent: 'user-agent',
-    Inspector: 'inspector',
-    Regular: 'regular',
+  global.Protocol = {
+    CSS: {
+      StyleSheetOrigin: {
+        Injected: 'injected',
+        UserAgent: 'user-agent',
+        Inspector: 'inspector',
+        Regular: 'regular',
+      },
+    },
   };
 
-  global.CSS = {};
-  global.CSS.supports = () => true;
-
-  // Stash the real one so we can reinstall after DT incorrectly polyfills.
-  // See https://github.com/GoogleChrome/lighthouse/issues/73
-  const _setImmediate = global.setImmediate;
-
-  global.WebInspector = {};
-  const WebInspector = global.WebInspector;
-
-  // Shared Dependencies
-  require('chrome-devtools-frontend/front_end/platform/utilities.js');
+  /**
+   * The single prototype augmentation needed from 'chrome-devtools-frontend/front_end/platform/utilities.js'.
+   * @return {Array<number>}
+   */
+  String.prototype.computeLineEndings = function() { // eslint-disable-line no-extend-native
```

Comment:
seemed simpler (and more up-front about what we're doing) to explicitly provide the single augmentation we needed rather than all of `utilities.js`. Since we're up to date now, maybe we could upstream putting this on TextUtils or something, instead :)